### ISSUE TITLE
ImapClient.ConnectAsync fails in v4.2.0: Syntax error in CAPABILITIES. Unexpected token: '+'

### ISSUE THREAD

**Issue by dsuesstr** [jstedfast/MailKit#1654]
The ImapClient fails on ConnectAsync  after we updated the nuget from 4.1.0 to 4.2.0 with the message:

_Syntax error in CAPABILITIES. Unexpected token: '+'_

We tested with different servers (e.g. outlook.office365.com)

**Platform**
 - OS: Windows
 - .NET Runtime: CoreCLR
 - .NET Framework: .NET 6
 - MailKit Version: 4.2

**Exception**
Syntax error in CAPABILITIES. Unexpected token: '+' ---> MailKit.Net.Imap.ImapProtocolException: Syntax error in CAPABILITIES. Unexpected token: '+'
ImapEngine.UpdateCapabilitiesAsync(ImapTokenType sentinel, CancellationToken cancellationToken)
ImapEngine.ProcessUntaggedResponseAsync(CancellationToken cancellationToken)
ImapCommand.StepAsync()
ImapEngine.IterateAsync()
ImapEngine.RunAsync(ImapCommand ic)
ImapClient.ConnectAsync(String host, Int32 port, SecureSocketOptions options, CancellationToken cancellationToken)
ImapClient.CreateImapClientAsync(CancellationToken cancellationToken) line 128

**Code Snippets**

```csharp
using var imapClient = new ImapClient();
var port = _config.Port; // 993
var encryption = _config.Encryption; // SslOnConnect

await _imapClient.ConnectAsync(_config.Address, port, encryption, cancellationToken); // <-- throws exception
// ...
```

**Comment by dsuesstr** [comment]
here's the log (Replaced the hostname with <ommtted>

```
Connected to imaps://<omittted>:993/
S: * OK IMAP4 ready
C: A00000000 CAPABILITY
S: * CAPABILITY IMAP4rev1 UIDPLUS IDLE LITERAL + QUOTA AUTH=PLAIN AUTH=LOGIN
S: A00000000 OK completed
```

**Comment by jstedfast** [comment]
Does this happen to be imaps://mail.infomaniak.com:993/ like in the other bug report I just marked as a duplicate of this one?

`LITERAL +` is a server bug because it *SHOULD* be `LITERAL+` (i.e. no space).

That said, I'm sure I can whip up a fix for you guys.

**Comment by jstedfast** [comment]
In about an hour (maybe less), https://www.myget.org/feed/mimekit/package/nuget/MailKit should have a v4.2.0.836 that you can install to test this fix.

Let me know if you run into any other issues.

Thanks!

**Comment by jpmaistrebazin** [comment]
No we get a new exception

System.InvalidCastException: Unable to cast object of type 'System.Char' to type 'System.String'.
   at MailKit.Net.Imap.ImapEngine.UpdateCapabilitiesAsync(ImapTokenType sentinel, CancellationToken cancellationToken)
   at MailKit.Net.Imap.ImapEngine.ProcessUntaggedResponseAsync(CancellationToken cancellationToken)
   at MailKit.Net.Imap.ImapCommand.StepAsync()
   at MailKit.Net.Imap.ImapEngine.IterateAsync()
   at MailKit.Net.Imap.ImapEngine.RunAsync(ImapCommand ic)
   at MailKit.Net.Imap.ImapClient.PostConnectAsync(Stream stream, String host, Int32 port, SecureSocketOptions options, Boolean starttls, CancellationToken cancellationToken)
   at MailKit.Net.Imap.ImapClient.ConnectAsync(String host, Int32 port, SecureSocketOptions options, CancellationToken cancellationToken)
   at RsSolutions.Motor.Test.EmailTest.Connect_Test() in /Users/jean-philippe/_work/rs-motor/rs-motor-api/RsSolutions.Motor.Test/EmailTest.cs:line 21

**Comment by jstedfast** [comment]
d'oh, thanks.

**Comment by jstedfast** [comment]
Just pushed a fix for that. It should be v4.2.0.837 when it gets uploaded to myget

**Comment by jpmaistrebazin** [comment]
It works

**Comment by dsuesstr** [comment]
@jstedfast thanks for the fix (or workaround). works for us too. 
And yes, it is the same mail host (infomaniak) :). I'll also create a ticket with them.

**Comment by jstedfast** [comment]
This was released as part of MailKit v4.3.0
